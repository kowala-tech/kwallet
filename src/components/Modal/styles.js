const styles = () => ({
	content: {
		display: "flex",
		padding: 0
	},
	contentText: {
		display: "flex",
		flex: 1
	},
	flex: {
		flex: 1,
	},
	menuButton: {
		marginLeft: 20,
		marginRight: -12,
	},
});

export default styles;
