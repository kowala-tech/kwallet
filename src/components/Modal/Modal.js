import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import withMobileDialog from "@material-ui/core/withMobileDialog";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
// Component Related Imports
import styles from "./styles";

class Modal extends React.Component {
	render() {

		const {
			open,
			handleClose,
			title,
			content,
			acceptButton,
			rejectButton,
			fullScreen,
			classes
		} = this.props;

		return (
			<Dialog
				open={open}
				onClose={handleClose}
				onBackdropClick={handleClose}
				fullScreen={fullScreen}
			>
				<AppBar position="static">
					<Toolbar>
						<Typography
							variant="title"
							color="inherit"
							className={classes.flex}
						>
							{title}
						</Typography>
						<IconButton
							onClick={handleClose}
							className={classes.menuButton}
							color="inherit"
						>
							<CloseIcon />
						</IconButton>
					</Toolbar>
				</AppBar>
				<DialogContent classes={{ root: classes.content }}>
					{content}
				</DialogContent>
				<DialogActions>
					{rejectButton}
					{acceptButton}
				</DialogActions>
			</Dialog>
		);
	}
}

Modal.propTypes = {
	classes: PropTypes.object.isRequired,
	open: PropTypes.bool.isRequired,
	handleClose: PropTypes.func.isRequired,
	title: PropTypes.string.isRequired,
	content: PropTypes.object.isRequired,
	acceptButton: PropTypes.object,
	rejectButton: PropTypes.object,
	fullScreen: PropTypes.bool.isRequired
};

export default compose(
	withStyles(styles),
	withMobileDialog()
)(Modal);
