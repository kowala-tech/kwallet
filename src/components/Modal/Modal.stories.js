import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
// Story related component imports
import Modal from "./Modal";
import Button from "@material-ui/core/Button";

storiesOf("Modal", module)
	.add("with example content", () =>
		(<Modal
			open={true}
			title={"Example Modal"}
			content={"This is an example modal used for displaying content over an the current page."}
			rejectButton={(
				<Button
					onClick={action("onReject")}
				>
					No means no!
				</Button>
			)}
			acceptButton={(
				<Button
					variant="raised"
					color="secondary"
					onClick={action("onAccept")}
				>
					Thanks for the rad info brosef!
				</Button>
			)}
			handleClose={action("onReject")}
		/>)
	);
