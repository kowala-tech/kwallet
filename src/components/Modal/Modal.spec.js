import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import Modal from "./Modal";
import Button from "@material-ui/core/Button";

describe("Modal", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<Modal
					open={true}
					title={"Modal Title"}
					content={"Modal content."}
					rejectButton={(
						<Button
							onClick={jest.fn()}
						>
							No
						</Button>
					)}
					acceptButton={(
						<Button
							onClick={jest.fn()}
						>
							Yes
						</Button>
					)}
					handleClose={jest.fn()}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
