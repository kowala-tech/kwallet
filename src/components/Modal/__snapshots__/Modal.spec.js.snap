// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal it should match the snapshot 1`] = `
<WithTheme(WithWidth(WithMobileDialog))
  acceptButton={
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      Yes
    </WithStyles(Button)>
  }
  classes={
    Object {
      "content": "WithTheme-WithWidth-WithMobileDialog---content-1",
      "contentText": "WithTheme-WithWidth-WithMobileDialog---contentText-2",
      "flex": "WithTheme-WithWidth-WithMobileDialog---flex-3",
      "menuButton": "WithTheme-WithWidth-WithMobileDialog---menuButton-4",
    }
  }
  content="Modal content."
  handleClose={[MockFunction]}
  open={true}
  rejectButton={
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      No
    </WithStyles(Button)>
  }
  title="Modal Title"
/>
`;
